### ISSUE TITLE
Add Create UserGroups

### ISSUE THREAD

**Issue by catdevman** [catdevman/terraform-provider-clickup#8]
- [ ] Need to add create (maybe delete and update as well???) User Group [here](https://github.com/raksul/go-clickup/blob/main/clickup/user_groups.go)
- [ ] Update this provider to user new version of go-clickup with create for user group
- [ ] Add User Group resource with create (maybe delete and update if available in go-clickup)

**Comment by rek** [comment]
The PR fort he GO CLI is ready.. Making some progress :)

**Comment by rek** [comment]
Ok the PR is merged upstream:
> - [x] Need to add create (maybe delete and update as well???) User Group [here](https://github.com/raksul/go-clickup/blob/main/clickup/user_groups.go)

**Comment by catdevman** [comment]
@rek looking good, let me know if you have any questions along the way... Of course I'm just learning terraform providers as well but anything I can help with to debug or review I'm happy to do. Appreciate you working on the provider. I hope you can get some use from it.

**Comment by rek** [comment]
This has been pretty fun actually. And I will for sure be using this at work, manual group management is getting out of control!

This part i'm actually vague on:
>  Update this provider to user new version of go-clickup

I should have a draft PR coming soon, I have the data-source working at least now

**Comment by catdevman** [comment]
Yeah I just meant updating the version of go-clickup once they merged and put out a new version

https://github.com/catdevman/terraform-provider-clickup/blob/af5d1a09c4c2d99456ccd6b5db69e30f03db1bcc/go.mod#L11

**Comment by rek** [comment]
Ahh gotcha, thanks! Have done. 

Ill make a draft PR soon just to keep track of the progress. I hope to finish it this week.

Edit: https://github.com/catdevman/terraform-provider-clickup/pull/13

**Comment by rek** [comment]
Ok I think we are close now. I made a small [PR](https://github.com/catdevman/terraform-provider-clickup/pull/14) to make the main branch green first.
